pkg/agent/deployer/generic/generic.go

```diff
@@ -226,3 +228,98 @@ func (deployer *Deployer) AddController(gvk schema.GroupVersionKind, controller
 		deployer.rsControllers[gvk] = controller
 	}
 }
+
+func (deployer *Deployer) SyncDescriptionStatus(clusternetClient *clusternetclientset.Clientset, dynamicClient dynamic.Interface,
+	restMapper meta.RESTMapper, desc *appsapi.Description) error {
+	objectsToBeDeployed := desc.Spec.Raw
+	Status := desc.Status.DeepCopy()
+	for _, object := range objectsToBeDeployed {
+		resource := &unstructured.Unstructured{}
+		err := resource.UnmarshalJSON(object)
+		if err != nil {
+			msg := fmt.Sprintf("failed to unmarshal resource: %v", err)
+			klog.ErrorDepth(5, msg)
+			continue
+		}
+
+		restMapping, err := restMapper.RESTMapping(resource.GroupVersionKind().GroupKind(), resource.GroupVersionKind().Version)
+		if err != nil {
+			klog.Errorf("please check whether the advertised apiserver of current child cluster is accessible. %v", err)
+			return err
+		}
+
+		// Get cluster Resource
+		resourceObj, err := dynamicClient.Resource(restMapping.Resource).Namespace(resource.GetNamespace()).
+			Get(context.TODO(), resource.GetName(), metav1.GetOptions{})
+		if err != nil {
+			msg := fmt.Sprintf("failed to Get resource: %v", err)
+			klog.ErrorDepth(5, msg)
+			return err
+		}
+		manifestStatus := appsapi.ManifestStatus{
+			Feed: appsapi.Feed{
+				Kind:       resourceObj.GetKind(),
+				APIVersion: resourceObj.GetAPIVersion(),
+				Namespace:  resourceObj.GetNamespace(),
+				Name:       resourceObj.GetName(),
+			},
+		}
+		statusMap, _, err := unstructured.NestedMap(resourceObj.Object, "status")
+		if err != nil {
+			klog.Errorf("Failed to get status field from %s(%s/%s), error: %v", resourceObj.GetKind(),
+				resourceObj.GetNamespace(), resourceObj.GetName(), err)
+			return err
+		}
+		result := &unstructured.Unstructured{}
+		result.SetUnstructuredContent(statusMap)
+
+		manifestStatus.FeedStatus.Reset()
+		manifestStatus.FeedStatus.Object = result.DeepCopyObject()
+
+		// update exit feed manifeststatus
+		feedExit := false
+		for index, status := range Status.ManifestStatuses {
+			if status.Feed.Namespace == manifestStatus.Feed.Namespace &&
```

Comment:
Here, reduce time complexity.
Parse the `ManifestStatuses` into a `map[namespacedname] int32, before range objectsToBeDeployed. Like:  {"default/cm1": 0, "default/deploy1":1 }
Much easier to check, if feed status existed, just get index and update it. If not exist then append.

Reply:
> Here, reduce time complexity. Parse the `ManifestStatuses` into a `map[namespacedname] int32, before range objectsToBeDeployed. Like: {"default/cm1": 0, "default/deploy1":1 } Much easier to check, if feed status existed, just get index and update it. If not exist then append.

Good idea, I wil try.